### PR TITLE
migration of login component (home, navbar, activate, reset finish, register)

### PR DESCRIPTION
**Pull request by flaviencathala**

### CODE REVIEW

File: generators/client-2/templates/src/main/webapp/app/account/_account.module.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ import { SettingsComponent } from "./settings/settings.component";
 
 <% if (enableTranslation) { %>upgradeAdapter.upgradeNg1Provider('$translate');<% } %>
 upgradeAdapter.upgradeNg1Provider('Auth');
-upgradeAdapter.upgradeNg1Provider('LoginService');
+//upgradeAdapter.upgradeNg1Provider('LoginService');
```
Comment (wmarques) [inline generators/client-2/templates/src/main/webapp/app/account/_account.module.ts]: I think you can remove this line.

File: generators/client-2/templates/src/main/webapp/app/components/_common.module.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ import { AuthServerProvider } from './auth/auth-jwt.service';
 import { AuthServerProvider } from './auth/auth-session.service';
 <%_ } _%>
 import { Account } from './auth/account.service';
-import { LoginService } from './login/login.service';
+//import { LoginService } from './login/login.service';
```
Comment (wmarques) [inline generators/client-2/templates/src/main/webapp/app/components/_common.module.ts]: same, remove commented lines in this file as we import the login service in appmodule now.

**Comment by wmarques** [comment]
@flaviencathala You have conflicts, please rebase your branch to the current `angular-2` branch.

**Comment by flaviencathala** [comment]
Done !

**Comment by ruddell** [comment]
Didn't test it but the changes look good.  For Travis, you need to fix `test-creation.js` to check for the component instead of the controller.

**Comment by wmarques** [comment]
@flaviencathala Fix test and cleanup a bit the code by removing useless comments and I'll merge this :)

File: generators/client-2/templates/src/main/webapp/app/account/_account.ng2module.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { NgModule } from '@angular/core';
 import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
 import { UIRouterModule } from "ui-router-ng2";
+import { BrowserModule } from '@angular/platform-browser';
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/_account.ng2module.ts]: why is this added? its already added in the root module

File: generators/client-2/templates/src/main/webapp/app/account/_account.ng2module.ts [hunk 4]
```diff
@@ -46,10 +47,12 @@ let ACCOUNT_STATES = [
 
 @NgModule({
     imports: [
+        BrowserModule,
         <%=angular2AppName%>SharedModule,
         UIRouterModule.forChild({ states: ACCOUNT_STATES })
     ],
     declarations: [
+        //<%=jhiPrefixCapitalized%>LoginModalComponent,
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/_account.ng2module.ts]: this shouldn't be imported here it should be imported in the common module and exported from there so that it can be used in all other modules

File: generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts [hunk 5]
```diff
@@ -26,12 +27,12 @@ import { ActiveMenuDirective } from './layouts/navbar/active-menu.directive';
         <%=angular2AppName%>AccountModule
     ],
     declarations: [
+        <%=jhiPrefixCapitalized%>LoginModalComponent,
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts]: this shouldn't be imported here it should be imported in the common module and exported from there so that it can be used in all other modules

File: generators/client-2/templates/src/main/webapp/app/account/_account.ng2module.ts [hunk 6]
```diff
@@ -46,10 +47,12 @@ let ACCOUNT_STATES = [
 
 @NgModule({
     imports: [
+        BrowserModule,
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/_account.ng2module.ts]: plz remove this its not required here

File: generators/client-2/templates/src/main/webapp/app/account/activate/_activate.component.ts [hunk 7]
```diff
@@ -1,16 +1,22 @@
...
+    Auth: any;
+    $stateParams: any;
+    modalRef: NgbModalRef;
 
-    constructor(@Inject('Auth') private Auth, @Inject('LoginService') private LoginService, @Inject('$stateParams') private $stateParams) {
-        this.login = this.LoginService.open;
+    constructor(@Inject('Auth') Auth,
+                private modalService: NgbModal,
+                @Inject('$stateParams') $stateParams) {
+        this.Auth = Auth;
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/activate/_activate.component.ts]: not required just add private to the argument declaration in the constructor

File: generators/client-2/templates/src/main/webapp/app/account/activate/_activate.component.ts [hunk 8]
```diff
@@ -23,4 +29,14 @@ export class ActivateComponent implements OnInit {
         });
     }
 
+    login(template) {
+        console.log('login activate');
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/activate/_activate.component.ts]: remove this console log

File: generators/client-2/templates/src/main/webapp/app/account/password-reset/finish/_password-reset-finish.component.ts [hunk 9]
```diff
@@ -14,14 +14,19 @@ export class PasswordResetFinishComponent implements OnInit {
...
+    $stateParams: any;
 
-    constructor(@Inject('Auth') private Auth, @Inject('LoginService') private LoginService,
-                @Inject('$stateParams') private $stateParams,
-                private elementRef: ElementRef, private renderer: Renderer) {}
+    constructor(@Inject('Auth') Auth,
+                @Inject('$stateParams') $stateParams,
+                private elementRef: ElementRef,
+                private renderer: Renderer) {
+        this.Auth = Auth;
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/password-reset/finish/_password-reset-finish.component.ts]: same here use private arguments

File: generators/client-2/templates/src/main/webapp/app/account/password-reset/finish/_password-reset-finish.component.ts [hunk 10]
```diff
@@ -14,14 +14,19 @@ export class PasswordResetFinishComponent implements OnInit {
...
+        this.$stateParams = $stateParams;
+    }
 
     ngOnInit() {
         this.resetAccount = {};
-        this.login = this.LoginService.open;
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/password-reset/finish/_password-reset-finish.component.ts]: how is this handled

File: generators/client-2/templates/src/main/webapp/app/components/_common.ng2module.ts [hunk 11]
```diff
@@ -10,9 +11,11 @@ import { jhiAlertComponent } from "./alert/alert.component";
 <%_ if (enableTranslation){ _%>
 import { <%=jhiPrefixCapitalized%>LanguageService } from './language/language.service';
 <%_ } _%>
+//import {LoginComponent} from "./login/login.component";
 
 @NgModule({
     imports: [
+        BrowserModule,
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/components/_common.ng2module.ts]: why?

File: generators/client-2/templates/src/main/webapp/app/components/login/_login.component.ts [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+            username: null,
+            password: null,
+            rememberMe: true
+        };
+        this.authenticationError = false;
+        this.modalRef.dismiss('cancel');
+    }
+
+    login () {
+        let vm = this;
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/components/login/_login.component.ts]: not required use arrow functions to access `this` directly

File: generators/client-2/templates/src/main/webapp/app/components/login/_login.component.ts [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+            rememberMe: this.rememberMe
+        }).then(function () {
+            vm.authenticationError = false;
+            vm.modalRef.dismiss('cancel');
+            if (vm.$state.current.name === 'register' || vm.$state.current.name === 'activate' ||
+                vm.$state.current.name === 'finishReset' || vm.$state.current.name === 'requestReset') {
+                vm.$state.go('home');
+            }
+
+            //vm.$rootScope.$broadcast('authenticationSuccess');
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/components/login/_login.component.ts]: we need to emit an observable here and subscribe that event

File: generators/client-2/templates/src/main/webapp/app/components/login/login.html [hunk 14]
```diff
@@ -1,6 +1,8 @@
 <div class="modal-header">
-    <button type="button" class="close" data-dismiss="modal" aria-hidden="true"
-            ng-click="vm.cancel()">&times;</button>
+    <button aria-label="Close" data-dismiss="modal" class="close" type="button" (click)="dismiss('closed')"><span aria-hidden="true">x</span>
+    </button>
+    <!--<button type="button" class="close" data-dismiss="modal" aria-hidden="true"
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/components/login/login.html]: remove

File: generators/client-2/templates/src/main/webapp/app/components/login/login.html [hunk 15]
```diff
@@ -9,51 +11,37 @@ <h4 class="modal-title" translate="login.title">Sign in</h4>
             <h1 translate="login.title">Sign in</h1>
         </div>
         <div class="col-md-8 col-md-offset-2">
-            <div class="alert alert-danger" ng-show="vm.authenticationError" translate="login.messages.error.authentication">
+            <div class="alert alert-danger" *ngIf="authenticationError" translate="login.messages.error.authentication">
                 <strong>Failed to sign in!</strong> Please check your credentials and try again.
             </div>
         </div>
-        <%_ if (enableSocialSignIn) { _%>
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/components/login/login.html]: this shouldnt be removed

File: generators/client-2/templates/src/main/webapp/app/components/login/login.html [hunk 16]
```diff
@@ -9,51 +11,37 @@ <h4 class="modal-title" translate="login.title">Sign in</h4>
...
             <div class="alert alert-warning">
-                <a class="alert-link" href="" ng-click="vm.requestResetPassword()" translate="login.password.forgot">Did you forget your password?</a>
+                <a class="alert-link" href="" (click)="requestResetPassword()" translate="login.password.forgot">Did you forget your password?</a>
             </div>
             <div class="alert alert-warning" translate="global.messages.info.register" translate-compile>
-                You don't have an account yet? <a class="alert-link" href="" ng-click="vm.register()">Register a new account</a>
+                You don't have an account yet? <a class="alert-link" href="" (click)="register()">Register a new account</a>
             </div>
         </div>
-        <%_ if (enableSocialSignIn) { _%>
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/components/login/login.html]: shouldnt be removed

File: generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html [hunk 17]
```diff
@@ -72,8 +72,8 @@
                                 <span translate="global.menu.account.logout">Sign out</span>
                             </a>
                         </li>
-                        <li uiSrefActive="active" *ngSwitchCase="false">
-                            <a href="" (click)="login()" id="login">
+                        <li ui-sref-active="active" *ngSwitchCase="false">
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/layouts/navbar/navbar.html]: remove `ui-sref-active="active"` it not required

File: generators/client-2/templates/src/main/webapp/app/account/activate/_activate.component.ts [hunk 18]
```diff
@@ -23,4 +29,14 @@ export class ActivateComponent implements OnInit {
...
+        this.modalRef = this.modalService.open(template);
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/activate/_activate.component.ts]: we need a better way for this. lets just use the loginService like before and do the result handling within that and return the modelRef from there

File: generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts (hunk 5 above)
Comment (flaviencathala) [inline generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts]: Do you mean export it from app module and then import it in account module from the app module ? What about home and navbar components do i import it directly in the components ?

Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts]: no just declare `LoginModalComponent` and then export it from the `common.ng2module` as below, now import the `<%=angular2AppName%>CommonModule` in the `account.ng2module` and you should be able to access any exported members from there. home and navbar are not reused hence they can be in the `app.ng2module`
```
@NgModule({
    imports: [
        <%=angular2AppName%>SharedModule
    ],
    declarations: [
        jhiAlertComponent,
        PageRibbonComponent,
        <%=jhiPrefixCapitalized%>LoginModalComponent
    ],
    providers: [
        AlertService,
        ProfileService,<%_ if (enableTranslation){ _%>
        <%=jhiPrefixCapitalized%>LanguageService,
        <%_ } _%>
        Account,
        Principal,
        AuthService
    ],
    exports: [
        <%=jhiPrefixCapitalized%>LoginModalComponent
    ],
    schemas: [CUSTOM_ELEMENTS_SCHEMA]

})
export class <%=angular2AppName%>CommonModule {}
```

File: generators/client-2/templates/src/main/webapp/app/account/_account.module.ts [hunk 19]
```diff
@@ -39,6 +40,7 @@ angular
         'ui.bootstrap',
         'ui.router'
     ])
+    .directive('<%=jhiPrefix%>LoginModal', <angular.IDirectiveFactory> upgradeAdapter.downgradeNg2Component(<%=jhiPrefixCapitalized%>LoginModalComponent))
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/_account.module.ts]: I dont think its required here. If needed do it in the common module

**Comment by deepu105** [comment]
please also rebase against latest branch as there are lot of changes in auth services

File: generators/client-2/templates/src/main/webapp/app/components/login/login.html (hunk 14 above)
Comment (flaviencathala) [inline generators/client-2/templates/src/main/webapp/app/components/login/login.html]: If i remove it the only way to close the window will be to click outside of the modal. I think the cross on the top right corner is very useful.

Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/components/login/login.html]: I was talking about the commented out code, remove commented code

Comment (flaviencathala) [inline generators/client-2/templates/src/main/webapp/app/components/login/login.html]: Ah ok

File: generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts (hunk 5 above)
Comment (flaviencathala) [inline generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts]: It works perfectly, thanks !

File: generators/client-2/templates/src/main/webapp/app/components/login/_login.component.ts (hunk 13 above)
Comment (flaviencathala) [inline generators/client-2/templates/src/main/webapp/app/components/login/_login.component.ts]: It throws me an error ("No provider for $rootScope!") so i commented it the time i found a solution. Did you have some problems when injecting $rootScope in the alert-error module ? Thanks.

**Comment by flaviencathala** [comment]
@deepu105 
I added the result handling in the loginService, did all the small modifications you asked and rebased.
It works for home screen, navbar, activate and reset password screens.
It fails on register screen, and the injection of $rootScope in login component still throws me an error.
I'm on it.

**Comment by flaviencathala** [comment]
@deepu105 The $rootScope error has been fixed with the rebase, could you please merge with angular-2 if it's ok for you ? Thanks.
